include/boost/process/v2/impl/exe.ipp

```diff
@@ -0,0 +1,324 @@
+#define BOOST_PROCESS_V2_IMPL_EXE_IPP
+
+#include <boost/process/v2/detail/config.hpp>
+#include <boost/process/v2/detail/last_error.hpp>
+#include <boost/process/v2/detail/throw_error.hpp>
+#include <boost/process/v2/detail/exe.hpp>
+#include <boost/process/v2/exe.hpp>
+
+#include <string>
+#include <vector>
+#include <sstream>
+
+#if defined(BOOST_PROCESS_V2_WINDOWS)
+#include <windows.h>
+#else
+#include <climits>
+#endif
+
+#if (defined(__APPLE__) && defined(__MACH__))
+#include <sys/proc_info.h>
+#include <libproc.h>
+#endif
+
+#if (defined(BOOST_PROCESS_V2_WINDOWS) || defined(__linux__) || defined(__ANDROID__) || defined(__sun))
+#include <cstdlib>
+#endif
+
+#if (defined(__FreeBSD__) || defined(__DragonFly__) || defined(__NetBSD__))
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#endif
+
+#if defined(__OpenBSD__)
+#include <sys/types.h>
+#include <sys/param.h>
+#include <sys/sysctl.h>
+#include <kvm.h>
+#endif
+
+#ifdef BOOST_PROCESS_USE_STD_FS
+namespace filesystem = std::filesystem;
+#else
+namespace filesystem = boost::filesystem;
+#endif
+
+BOOST_PROCESS_V2_BEGIN_NAMESPACE
+
+namespace ext {
+
+#if defined(BOOST_PROCESS_V2_WINDOWS)
+
+filesystem::path exe_path(boost::process::v2::pid_type pid, std::error_code & ec)
+{
+    filesystem::path path;
+    if (pid == GetCurrentProcessId()) 
+    {
+        wchar_t buffer[MAX_PATH];
+        if (GetModuleFileNameW(nullptr, buffer, sizeof(buffer)) != 0) 
+        {
+            wchar_t exe[MAX_PATH];
+            if (_wfullpath(exe, buffer, MAX_PATH)) 
+            {
+                path = exe;
+            }
+        }
+    } 
+    else 
+    {
+        HANDLE proc = detail::ext::open_process_with_debug_privilege(pid, ec);
+        if (proc == nullptr) return path;
+        wchar_t buffer[MAX_PATH];
+        DWORD size = sizeof(buffer);
+        if (QueryFullProcessImageNameW(proc, 0, buffer, &size) != 0) 
+        {
+            wchar_t exe[MAX_PATH];
+            if (_wfullpath(exe, buffer, MAX_PATH)) 
+            {
+                path = exe;
+            }
+        }
+        CloseHandle(proc);
+    }
+    if (path.string().empty())
+        ec = detail::get_last_error();
+    return path;
+}
+
+#elif (defined(__APPLE__) && defined(__MACH__))
+
+filesystem::path exe_path(boost::process::v2::pid_type pid, std::error_code & ec)
+{
+    std::string path;
+    char exe[PROC_PIDPATHINFO_MAXSIZE];
+    if (proc_pidpath(pid, exe, sizeof(exe)) > 0) 
+    {
+        char buffer[PATH_MAX];
+        if (realpath(exe, buffer)) 
+        {
+            path = buffer;
+        }
+    }
+    if (path.empty())
+        ec = detail::get_last_error();
+    return path;
+}
+
+#elif (defined(__linux__) || defined(__ANDROID__))
+
+filesystem::path exe_path(boost::process::v2::pid_type pid, std::error_code & ec)
+{
+    std::string path;
+    char exe[PATH_MAX];
+    if (realpath(("/proc/" + std::to_string(pid) + "/exe").c_str(), exe)) 
+    {
+        path = exe;
+    }
+    if (path.empty())
```

Comment:
This error handling probably works, but is incorrect. Doc states

> Otherwise, it returns NULL, the contents of the array  resolved_path are undefined, and [errno](https://man7.org/linux/man-pages/man3/errno.3.html) is set to indicate the error.

So the way I'd write this is as so:

```cpp
    char exe[PATH_MAX];
    if (realpath(("/proc/" + std::to_string(pid) + "/exe").c_str(), exe) != nullptr) 
        path = exe;
    else
        ec = detail::get_last_error();
```